host/src/main.rs

```diff
@@ -36,78 +39,54 @@ pub struct Opt {
-    max_log_days: usize,
-
-    #[structopt(long, require_equals = true, default_value = "20")]
-    // WARNING: must be larger than concurrency_limit
-    max_caches: usize,
+    #[structopt(long, require_equals = true, default_value = "host/config/config.json")]
+    /// Path to a config file that includes sufficent json args to request
+    /// a proof of specified type. Curl json-rpc overrides its contents
+    config_path: PathBuf,
 
     #[structopt(long, require_equals = true)]
-    config_path: Option<PathBuf>,
+    /// Use a local directory as a cache for input. Accepts a custom directory.
+    cache: Option<PathBuf>,
 
     #[structopt(long, require_equals = true, env = "RUST_LOG", default_value = "info")]
+    /// Set the log level
     log_level: String,
 }
 
 #[tokio::main]
 async fn main() -> Result<()> {
+    let opt = Opt::from_args();
     let config = get_config(None).unwrap();
-    let opt = Opt::deserialize(&config).unwrap();
-    println!("Start config: {:?}", opt);
+    println!("Start config: {:?}", config);
 
-    let subscriber_builder = tracing_subscriber::FmtSubscriber::builder()
+    let subscriber = tracing_subscriber::FmtSubscriber::builder()
         .with_env_filter(&opt.log_level)
-        .with_test_writer();
-    let _guard = match opt.log_path {
-        Some(ref log_path) => {
-            let file_appender = tracing_appender::rolling::Builder::new()
-                .rotation(tracing_appender::rolling::Rotation::DAILY)
-                .filename_prefix("raiko.log")
-                .max_log_files(opt.max_log_days)
-                .build(log_path)
-                .expect("initializing rolling file appender failed");
-            let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
-            let subscriber = subscriber_builder.json().with_writer(non_blocking).finish();
-            tracing::subscriber::set_global_default(subscriber).unwrap();
-            Some(_guard)
-        }
-        None => {
-            let subscriber = subscriber_builder.finish();
-            tracing::subscriber::set_global_default(subscriber).unwrap();
-            None
-        }
-    };
-
+        .with_test_writer()
+        .finish();
+    tracing::subscriber::set_global_default(subscriber).unwrap();
     serve(opt).await?;
     Ok(())
 }
 
 /// Gets the config going through all possible sources
-pub fn get_config(request_config: Option<Value>) -> Result<Value> {
+fn get_config(request_config: Option<Value>) -> Result<Value> {
     let mut config = Value::default();
     let opt = Opt::from_args();
+    println!("     cli_args: {:?}", opt);
 
     // Config file has the lowest preference
-    if let Some(config_path) = &opt.config_path {
-        let file = File::open(config_path)?;
-        let reader = BufReader::new(file);
-        let file_config: Value = serde_json::from_reader(reader)?;
-        merge(&mut config, &file_config);
-    };
+    let file = File::open(&opt.config_path)?;
```

Comment:
Config file now required to exist, expected behavior?

Reply:
yea cuz there is an explicit default value in `Opt` which points to `host/config/config.json` which contains sufficient information for an exemplary proof request.
`Opt` itself only has the necessary params to start the server.